### PR TITLE
Remove RegExp in SelectV2

### PR DESCRIPTION
**Pull request by ghivert**
**Component**:
SelectV2

**Description**:
Remove RegExp because it was causing a bug in the match. Replaced with an interspersing.

### CODE REVIEW

File: src/lib/components/selectv2/Selectv2.component.tsx [hunk 1]
```diff
@@ -60,7 +60,10 @@ const InternalOption = (props) => {
   const formatOptionLabel = () => {
     const label = props.data.label;
     const inputValue = props.selectProps.inputValue;
-    const parts = label.split(new RegExp(`(${inputValue})`, 'gi'));
+    const parts = label.split(inputValue).flatMap((item, index) => {
+      if (index % 2 === 1) return [inputValue, item]
+      return [item]
+    });
```
Comment (JBWatenbergScality) [inline src/lib/components/selectv2/Selectv2.component.tsx]: I don't think that this is correct.
For the following input: 'test-bla-test-bla', searching for test 
We expect `['', 'test', '-bla-', 'test', '-bla']` whereas with your implementation we get: `['', '-bla-', 'test', '-bla']`

Maybe simply do a `.flatMap((item) => [item, inputValue])` and remove the last entry

Comment (ghivert) [reply src/lib/components/selectv2/Selectv2.component.tsx]: I think it's correct, but I can do something simpler. :)

Comment (ghivert) [reply src/lib/components/selectv2/Selectv2.component.tsx]: Updated. :)

File: src/lib/components/selectv2/Selectv2.component.tsx [hunk 2]
```diff
@@ -212,6 +216,8 @@ export type SelectProps = {
   children?: React.ReactNode;
   defaultValue?: string;
   value?: string;
+  onFocus: () => void;
+  onBlur: () => void;
```
Comment (JBWatenbergScality) [inline src/lib/components/selectv2/Selectv2.component.tsx]:
```suggestion
  onFocus?: (e: TOBEDEFINEDEVENT) => void;
  onBlur?: (e: TOBEDEFINEDEVENT) => void;
```

**Review by JBWatenbergScality — APPROVED** [review]

**Pull request merged**